ansible/roles/host_setup/tasks/pin_kernel.yml

```diff
@@ -18,44 +18,59 @@
     filter: "ansible_kernel"
 
 - name: Check Kernel Version
-  ansible.builtin.fail:
+  ansible.builtin.debug:
     msg: >
       Wrong kernel Version found
-      [ {{ ansible_facts['kernel'] }} < {{ host_required_kernel }} ]
+      [ {{ ansible_facts['kernel'] }} != {{ host_required_kernel }} ]
       Resolve this issue before continuing.
   when:
-    - ansible_facts['kernel'] is version(host_required_kernel, '<')
+    - ansible_facts['kernel'] is version(host_required_kernel, '!=')
 
-- name: Pin kernel packages version
-  ansible.builtin.copy:
-    dest: "{{ apt_preferences }}/pin-kernel"
-    content: |
-      Package: linux-image-{{ ansible_facts['kernel'] }}
+- name: "Install specific kernel image/modules/extra version {{ host_required_kernel }}"
+  ansible.builtin.apt:
+    name:
+      - "linux-image-{{ host_required_kernel }}"
```

Comment:
Please make that multi distro ready while you work on this, and add overrides we can manipulate for different OS